api/cpp/include/slint-testing.h

```diff
@@ -25,7 +26,60 @@ class ElementHandle
 {
     cbindgen_private::ElementHandle inner;
 
+    explicit ElementHandle(const cbindgen_private::ElementHandle *inner) : inner(*inner) { }
+
 public:
+    /// Visit elements of a component and call the visitor to each of them.
+    ///
+    /// The visitor must be a functor that takes a ElementHandle as argument and return either void,
+    /// or something that can be converted to bool.
+    /// If the type can be converted to bool, then false means that the visitor will continue to be
+    /// called, and true means that the visitor will not be called anymore and the visitation stops.
+    /// This function then returns the return type. If the visitor never returns somethng, a default
+    /// constructed object is returned. For example, if the visitor return a std::optional<Foo>, the
+    /// iteration will continue as long as the visitor returns nullopt, but once it returns a value,
+    /// that value is returned from this function.
+    ///
+    /// Examples:
+    /// - TBD: examples where the visitor returns void, example when it returns true, example when
```

Comment:
Todo for this PR or later?

Reply:
I wanted to do it in this PR but then I forgot